src/containers/EventForm/components/EventForm/function.js

```diff
@@ -0,0 +1,42 @@
+import {
+  PeopleRepository,
+  ArtworksRepository,
+  OrganisationsRepository,
+  LocationsRepository,
+  EventsRepository,
+} from 'hackoss';
+import { FirebaseApp } from 'utils/firebase';
+
+export default async function createEvent(data) {
+  const pplRepo = new PeopleRepository(FirebaseApp);
+  const locRepo = new LocationsRepository(FirebaseApp);
+  const orgRepo = new OrganisationsRepository(FirebaseApp);
+  const awRepo = new ArtworksRepository(FirebaseApp, pplRepo);
+  const eventRepo = new EventsRepository(
+    FirebaseApp,
+    pplRepo,
+    locRepo,
+    orgRepo,
+    awRepo,
+  );
+  /* eslint-disable */
+  let { banner, venue, speakers, startTime, endTime, ...others } = data;
+  /* eslint-enable */
+  const aw = await awRepo.getArtworks();
+  const loc = await locRepo.getLocations();
+  const ppl = await pplRepo.getPeople();
+  const org = await orgRepo.getOrganisations();
+  startTime = new Date(startTime);
+  endTime = new Date(endTime);
+  const bannerId = aw.find(item => item.title === banner).id;
```

Comment:
Not sure if it can be used, but there is a `getArtwork` method that retrieves a single artwork given the ID. Same goes for all other collections.

Reply:
It's a filter for getting id.
For this step, I'm using name to filter id. Since I cannot find a way to show name but value will be set to id when doing select in the frontend.